docs: add website docs for ButtonGroup (#2722)
twilio-labs/paste: packages/paste-website/src/pages/components/button-group/index.mdx

      }
    }
    mdx(frontmatter: {slug: {eq: "/components/button-group/"}}) {
      fileAbsolutePath
      frontmatter {
        slug
        title
      }
      headings {
        depth
        value
      }
    }
    allAirtable(filter: {data: {Feature: {eq: "Button Group"}}}) {
      edges {
        node {
          data {
            Documentation
            Figma
            Design_committee_review
            Engineer_committee_review
            Code
            status
          }
        }
      }
    }
  }
`;

<NormalizedComponentHeader
  categoryRoute={SidebarCategoryRoutes.COMPONENTS}
  githubUrl="https://github.com/twilio-labs/paste/tree/main/packages/paste-core/components/button-group"
  storybookUrl="/?path=/story/components-button-group--attached"
  data={props.data}
  description={props.pageContext.frontmatter.description}
/>

---

<contentwrapper>

<PageAside data={props.data.mdx} />

<content>

<LivePreview
  noInline
  scope={{
    ButtonGroup,
    Button,
    PlusIcon,
    CheckboxCheckIcon,
    PopoverContainer,
    PopoverButton,
    Popover,
    Heading,
    Separator,
    Text,
  }}
  language="jsx"
>
  {BasicExample}
</LivePreview>

## Guidelines

> A Button Group is a set of 2-6 buttons. It is often used to visually group related buttons together. The buttons should be independent from each other, i.e. clicking one button does not change the state of another button.

Reviewer: Instead of "related buttons" could this say "Related actions"?

Author: Yeah!

Reviewer: ```suggestion
A Button Group is a way to bring a set of 2-6 buttons together, often visually, that have related actions. The buttons should be independent from each other, i.e. clicking one button does not change the state of another button.
```

Any Paste components that use [Button](/components/button), including [Popover](/components/popover), [Modal](/components/modal), and [Minimizable Dialog](/components/minimizable-dialog) can be put into a ButtonGroup.

### Button Group vs Radio Button Group

> Radio Button Groups should be used when the user needs to pick one option from the group. Radio Buttons cannot be used wihtin a Button Group.

Reviewer: Oops,  pressed something weird here about batching the suggestions, my bad!

> Radio Button Groups should be used when the user needs to pick one option from the group. Radio Buttons cannot be used wihtin a Button Group.

Reviewer: I wanted to say that I think we could get a little more specific here. 

"Radio button groups should be used when the user needs to select a single option from a group of mutually exclusive choices. Radio Buttons cannot be used within a Button Group."

## Examples

### Default

By default, the Buttons in a Button Group are displayed horizontally with a margin between them.

<LivePreview
  noInline
  scope={{
    ButtonGroup,
    Button,
    PlusIcon,
    CheckboxCheckIcon,
    PopoverContainer,
    PopoverButton,
    Popover,
    Heading,
    Separator,
    Text,
  }}
  language="jsx"
>
  {UnattachedExample}
</LivePreview>

### Attached

Use the `attached` prop to make the buttons visually connected.

<LivePreview
  noInline
  scope={{
    ButtonGroup,
    Button,
    PlusIcon,
    CheckboxCheckIcon,
    PopoverContainer,
    PopoverButton,
    Popover,
    Heading,
    Separator,
    Text,
  }}
  language="jsx"
>
  {BasicExample}
</LivePreview>

### Icon

> Use the icon variant when space is limited and the icons being used are easily recognizable (e.g., icons representing graph types). It is highly recommended to pair this variant with the [Tooltip](/components/tooltip) component to help ensure understandability.

Reviewer: For the example on the docs site, should we pair it with a tooltips since we highly recommend that?

Author: Totally! I'll update it.

### API

#### Installation

```bash
yarn add @twilio-paste/button-group - or - yarn add @twilio-paste/core
```

#### Usage

```jsx
import {ButtonGroup} from '@twilio-paste/core/button-group';
import {Button} from '@twilio-paste/core/button';

const Component = () => (
  <ButtonGroup attached>
    <Button variant="primary">First button</Button>
    <Button variant="secondary">Second button</Button>
    <Button variant="destructive">Destructive button</Button>
  </ButtonGroup>
);
```

#### Props

All the valid HTML attributes for `div` are supported including the following props:

| Prop      | Type                   | Description                                                                               | Default        |
| --------- | ---------------------- | ----------------------------------------------------------------------------------------- | -------------- |
| children  | `React.ReactElement[]` | The Buttons rendered in the ButtonGroup                                                   |                |
| attached? | `boolean`              | If the group of buttons are visually attached                                             | `false`        |

> | element?  | `string`               | Overrides the default element name to apply unique styles with the Customization Provider | 'BUTTON_GROUP' |

Author: We usually leave it out of props tables... idk why.....

---
title: Button Group
package: '@twilio-paste/button-group'

> description: A Button Group is a set of related Buttons

Reviewer: ```suggestion
description: A Button Group displays a set of related actions.
```

            Code
            status
          }
        }
      }
    }
  }
`;

<NormalizedComponentHeader
  categoryRoute={SidebarCategoryRoutes.COMPONENTS}
  githubUrl="https://github.com/twilio-labs/paste/tree/main/packages/paste-core/components/button-group"
  storybookUrl="/?path=/story/components-button-group--attached"
  data={props.data}
  description={props.pageContext.frontmatter.description}
/>

---

<contentwrapper>

<PageAside data={props.data.mdx} />

<content>

<LivePreview
  noInline
  scope={{
    ButtonGroup,
    Button,
    PlusIcon,
    ExportIcon,
    DeleteIcon,
    EditIcon,
  }}
  language="jsx"
>
  {BasicExample}
</LivePreview>

## Guidelines

A Button Group is a way to bring a set of 2-6 buttons together, often visually, that have related actions. The buttons should be independent from each other, i.e. clicking one button does not change the state of another button.

Any Paste components that use [Button](/components/button), including [Popover](/components/popover), [Modal](/components/modal), and [Minimizable Dialog](/components/minimizable-dialog) can be put into a ButtonGroup.

### Button Group vs Radio Button Group

> Radio Button Groups should be used when the user needs to pick one option from the group. Radio Buttons cannot be used within a Button Group.

Reviewer: @shleewhite let me know if this makes sense. I thought I had left a comment like this in my previous review, but I don't see it here anymore. I wanted to add a little further clarification between the components.
